src/msac.rs

```diff
@@ -159,6 +160,22 @@ pub struct MsacContext {
     ) -> c_uint,
 }
 
+impl Default for MsacContext {
+    fn default() -> Self {
+        Self {
+            buf_pos: ptr::null(),
+            buf_end: ptr::null(),
+            dif: Default::default(),
+            rng: Default::default(),
+            cnt: Default::default(),
+            allow_update_cdf: Default::default(),
+
+            #[cfg(all(feature = "asm", target_arch = "x86_64"))]
+            symbol_adapt16: rav1d_msac_decode_symbol_adapt_c,
```

Comment:
```suggestion
            symbol_adapt16: Rav1dMsacDSPContext::default().symbol_adapt16,
```

Just to centralize the logic.